routemaster/services/ingest.rb

```diff
@@ -13,27 +13,47 @@ class Ingest
       include Mixins::Assert
       include Mixins::Counters
 
-      def initialize(topic:, event:, queue:)
+      def initialize(topic:, event:, queue:, subscriber_name: nil)
         _assert(event.topic == topic.name)
         @topic = topic
         @event = event
         @queue = queue
+        @subscriber_name = subscriber_name
       end
 
       def call
-        data = Services::Codec.new.dump(@event)
-        @topic.subscribers.each do |s|
-          batch = Models::Batch.ingest(data: data, timestamp: @event.timestamp, subscriber: s)
+        publish_to_subscribers
+        update_counters
+        self
+      end
+
+      private
 
-          job = Models::Job.new(name: 'batch', args: batch.uid, run_at: batch.deadline)
-          @queue.push(job)
-          @queue.promote(job) if batch.full?
-        end
+      def data
+        @_data ||= Services::Codec.new.dump(@event)
+      end
 
+      def update_counters
         _counters.incr('events.published', topic: @topic.name)
         _counters.incr('events.bytes', topic: @topic.name, count: data.length)
         @topic.increment_count
-        self
+      end
+
+      def publish_to_subscribers
+        @topic.subscribers.each { |s| publish_to(s) if should_publish_to?(s) }
```

Comment:
If I'm reading this right: if the client passes an invalid subscriber name (one that is not subscribed to this topic), nothing will happen — the event will just be ignored?

If so, I'd suggest changing that to raise an error. The rest of `routemaster`'s codebase errs on the defensive side, which I like to think helps for a critical piece of software, and reduces the amount of "magic".

What do you think?

Reply:
Yep, that's the part I was on the borderline about what to do. This statement solves is tho:

> helps for a critical piece of software

Couldn't agree more. Will follow.